src/features/projectsV2/ProjectsContext.tsx

```diff
@@ -286,17 +286,22 @@ export const ProjectsProvider: FC<ProjectsProviderProps> = ({
     )
       return;
 
-    // Handle the case where a direct link requests a specific plant location (via URL query).
-    // This will update the ploc param based on the requestedPlantLocation. If the requested hid is invalid,
-    // it falls back to the default (first) site.
     if (requestedPlantLocation && selectedPlantLocation === null) {
-      const result = plantLocations?.find(
-        (plantLocation) => plantLocation.hid === requestedPlantLocation
-      );
-      if (result) {
-        setSelectedPlantLocation(result);
+      const hasNoSites = singleProject.sites?.length === 0;
+
+      if (hasNoSites) {
+        //Case when a direct link requests a specific plant location but no sites exist for a project(e.g projectSlug: mothersforest).
+        updateUrlWithSiteId(locale, singleProject.slug, null);
```

Comment:
This logic is a little confusing to me.

Why are we calling `updateUrlWithSiteId` directly? Why not pass in `null` to `updateSiteAndUrl`?